community/ndp/transport-socket/src/main/java/org/neo4j/ndp/transport/socket/SocketProtocolV1.java

```diff
@@ -70,7 +71,7 @@ public SocketProtocolV1( final Log log, Session session )
         this.session = session;
         this.output = new ChunkedOutput();
         this.input = new ChunkedInput();
-        this.packer = new PackStreamMessageFormatV1.Writer( output, output.messageBoundaryHook() );
+        this.packer = new PackStreamMessageFormatV1.Writer( new PackStream.Packer( output ), output.messageBoundaryHook() );
```

Comment:
Actually, reading the code, I change my mind, I think what davide is proposing here is cleaner - we're asking for a PackOutput already anyway, so the user will need to know about PackStream. I think we should merge this as it stands